doc/tutorials/enable_ptm.rst

```diff
@@ -0,0 +1,67 @@
+.. _enable-ptm:
+
+Enable PCIe Precision Time Management
+#####################################
+
+The PCI Express (PCIe) specification defines a Precision Time Measurement (PTM)
+mechanism that lets you coordinate and synchronize events across multiple PCI
+components within the same system with very fine time precision.
+
+ACRN adds PCIe root port emulation in the hypervisor to support the PTM feature
+and emulates a simple PTM hierarchy.  ACRN enables PTM in a Guest VM if the user
+sets the ``enable_ptm`` option when passing through a device to a post-launched
+VM. When the user enables PTM, the passthru device is connected to a virtual
+root port instead of the host bridge as it normally would.
+
+Here is an example launch script that enables PTM on a passthru Ethernet card:
+
+.. code-block:: bash
+   :emphasize-lines: 5
+
+   acrn-dm -A -m $mem_size -s 0:0,hostbridge \
+      -s 3,virtio-blk,uos-test.img \
+      -s 4,virtio-net,tap0 \
+      -s 5,virtio-console,@stdio:stdio_port \
+      -s 6,passthru,a9/00/0,enable_ptm \
+      --ovmf /usr/share/acrn/bios/OVMF.fd
+
+Here is the bus hierarchy from guest (as shown by the ``lspci`` command)::
+
+   sudo lspci -tv
+   -[0000:00]-+-00.0  Network Appliance Corporation Device 1275
+              +-03.0  Red Hat, Inc. Virtio block device
+              +-04.0  Red Hat, Inc. Virtio network device
+              +-05.0  Red Hat, Inc. Virtio console
+              \-06.0-[01]----00.0  Intel Corporation Device 15f2
+
+
+You can also verify that PTM was enabled by using ``dmesg`` in the guest VM::
+
+    $ dmesg | grep -i ptm
+    [ 1.555284] pci_ptm_init: 00:00.00, ispcie=1, type=0x4
+    [ 1.555356] Cannot find PTM ext cap.
+    [ 1.561311] pci_ptm_init: 00:03.00, ispcie=0, type=0x0
+    [ 1.567146] pci_ptm_init: 00:04.00, ispcie=0, type=0x0
+    [ 1.572983] pci_ptm_init: 00:05.00, ispcie=0, type=0x0
+    [ 1.718038] pci_ptm_init: 00:06.00, ispcie=1, type=0x4
+    [ 1.722034] ptm is ptm_root.
+    [ 1.723033] Condition-2: ptm is enabled.
+    [ 1.723052] pci 0000:00:06.0: PTM enabled (root), 4ns granularity
+    [ 1.766438] pci_ptm_init: a9:00.00, ispcie=1, type=0x0
+    [ 5.715000] igc_probe enable ptm.
```

Comment:
Is it expected that we seem to have a 4 sec delay here?

Reply:
I believe there are lots things happened during the 4 seconds.  The output doesn't contain everything (since I grep "ptm" from dmesg).  As far as whether enabling PTM slows down guest OS during boot, I don't know.